torch/_decomp/decompositions.py

```diff
@@ -579,6 +579,30 @@ def native_dropout_backward(grad_output: Tensor, mask: Tensor, scale: float):
     return grad_output * (mask.type_as(grad_output) * scale)
 
 
+@register_decomposition(aten.reciprocal)
+def reciprocal(self: Tensor) -> Tensor:
+    return 1 / self
+
+
+# Used for implementing ops that implicitly promote ints to floats. Could be in Primtorch refs?
```

Comment:
heh, it not just promotes int to float, it also makes sure that intermediate math is done in fp32? Logit core implementation does that.